runtime/src/swam/runtime/import_generator/ImportGenerator.scala

```diff
@@ -0,0 +1,182 @@
+package swam.runtime.import_generator
+
+import java.io.{File, PrintWriter}
+
+import swam.{FuncType, ValType}
+import swam.ValType.{F32, F64, I32, I64}
+import swam.runtime.Import
+import scala.reflect.runtime.universe._
+
+import sys.process._
+
+/**
+  * @author Javier Cabrera-Arteaga on 2020-03-06
+  */
+/**
+  * Generates the scala code with the template to use as Import in the WASM execution
+  */
+class ImportGenerator() {
+
+  val f = "IO"
+
+  /**
+    * Map WASM to scala primitive types
+    * @param v
+    * @return
+    */
+  def getScalaType(v: ValType): String = v match {
+    case I32 => s"Int"
+    case I64 => s"Long"
+    case F32 => s"Float"
+    case F64 => s"Double"
+  }
+
+  /**
+    * Creates the paramaters chain for the function signature in scala
+    * @param index
+    * @param tpr
+    * @return
+    */
+  def buildParameter(index: Int, tpr: ValType) =
+    s"p${tpr match {
+      case I32 => s"$index:${getScalaType(tpr)}"
+      case I64 => s"$index:${getScalaType(tpr)}"
+      case F32 => s"$index:${getScalaType(tpr)}"
+      case F64 => s"$index:${getScalaType(tpr)}"
+    }}"
+
+  /**
+    * Creates the string representing the scala function return based on the arity of the return object in WASM
+    * @param t
+    * @return
+    */
+  def buildReturn(t: Vector[ValType]) = {
+    if (t.isEmpty)
+      "Unit"
+    else {
+      if (t.length == 1)
+        getScalaType(t.last)
+      else
+        s"(${t.map(getScalaType).mkString(",")})"
+    }
+  }
+
+  var registeredFunctions: Set[String] = Set[String]()
+
+  /**
+    * Creates the function trait template
+    * @param moduleName
+    * @param fieldName
+    * @param func
+    * @return
+    */
+  def buildFunctionTemplate(moduleName: String, fieldName: String, func: FuncType): String = {
+    val name = s"${moduleName}_$fieldName"
+
+    s"def $name(${func.params
+      .zip(Seq.range(0, func.params.length))
+      .map(t => {
+        buildParameter(t._2, t._1)
+      })
```

Comment:
A better way of doing this is probably along these lines:
```suggestion
    s"def $name(${func.params
      .zipWithIndex
      .map {
        case (p, idx) => buildParameter(idx, p)
      }
```